velox/common/memory/MemoryArbitrator.cpp

```diff
@@ -60,34 +60,100 @@ class Registry {
   std::unordered_map<std::string, MemoryArbitrator::Factory> map_;
 };
 
-Registry registry;
+Registry& registry() {
+  static Registry registry;
+  return registry;
+}
+
+/// Used to enforce the fixed query memory isolation across running queries.
+/// When a memory pool exceeds the fixed capacity limit, the query just
+/// fails with memory capacity exceeded error without arbitration. This is
+/// used to match the current memory isolation behavior adopted by
+/// Prestissimo.
+///
+/// TODO: deprecate this legacy policy with kShared policy for Prestissimo
+/// later.
+class NoopArbitrator : public MemoryArbitrator {
+ public:
+  static void registerFactory() {
+    MemoryArbitrator::registerFactory(
+        kind_, [](const MemoryArbitrator::Config& config) {
+          return std::make_unique<NoopArbitrator>(config);
+        });
+  }
+
+  explicit NoopArbitrator(const Config& config) : MemoryArbitrator(config) {
+    VELOX_CHECK_EQ(kind_, config.kind);
+  }
+
+  std::string kind() override {
+    return kind_;
+  }
+
+  void reserveMemory(MemoryPool* pool, uint64_t bytes) override {
+    // NOTE: In noop arbitrator, we set the memory pool's
+    // capacity to its configured max once reserveMemory is called.
+    pool->grow(pool->maxCapacity());
+  }
+
+  void releaseMemory(MemoryPool* pool) override {
+    // No-op
+  }
+
+  bool growMemory(
+      MemoryPool* pool,
+      const std::vector<std::shared_ptr<MemoryPool>>& candidatePools,
+      uint64_t targetBytes) override {
+    // Growing is not allowed in noop arbitrator
```

Comment:
Shall we rephrase and put these comments as formal method comments to give better exposure?